lib/libutils/ext/include/util.h

```diff
@@ -188,7 +188,33 @@ static inline void reg_pair_from_64(uint64_t val, uint32_t *reg0,
 	*reg1 = low32_from_64(val);
 }
 
-/* Get and set bit fields  */
+/**
+ * Functions to get and set bit fields in a 32/64-bit register.
+ *
+ * These inline functions allow setting and extracting specific bits in
+ * a register @reg according to a given mask @mask. The mask
+ * specifies which bits in the register should be updated or extracted.
+ *
+ * - set_field*(): Modifies specific bits in a register by clearing the
+ *		   bits specified by the mask and then setting these
+ *		   bits to the new value @val.
+ * - get_field*(): Extracts the value of specific bits in a register by
+ *		   isolating the bits specified by the mask and then
+ *		   shifting them to the rightmost position.
+ *
+ * @reg The original 32-bit or 64-bit register value.
+ * @mask A bitmask indicating which bits in the register should be
+ * updated or extracted.
+ * @val The new value to be inserted into the register at the
+ * positions specified by the mask (only for set_field*() functions).
+ * These functions returns the updated register value with the specified
+ * bits set to the new value (for set_field*() functions), or the value
+ * of the bits specified by the mask, shifted to the rightmost position
+ * (for get_field*() functions).
```

Comment:
To make the description clearer, I would suggest to explicitly split the *get_field_uXX()* et *set_field_uXX()* descriptions:
```c
/*
 * These inline functions allow ...
 *
 * set_field_u32() / set_field_u64() - Modifies specific bits in ...
 * @reg: ...
 * @mask: ...
 * @val: ...
 * @return: ...
 *
 * get_field_u32() / get_field_u64() - Extracts the value of specific bits in 
 * @reg: ...
 * @mask: ...
 * @return: ...
 */
```
This is only a suggestion. In the end, what's important is that these information are as readable and clear as possible.